rclcpp/include/rclcpp/timer.hpp

```diff
@@ -198,6 +199,15 @@ class GenericTimer : public TimerBase
   void
   execute_callback() override
   {
+    // TODO(ivanpauno): Delete this workaround.
+    // The issue is that the multithreaded executor might schedule the timer
+    // more than once in different threads.
+    // Making `execute_callback` mutually exclusive and rechecking if the timer is ready
+    // avoids the issue.
+    std::lock_guard<std::mutex> lock(execution_mutex_);
```

Comment:
Is this related to #1212? It doesn't seem related to the intra process ring buffer being taken from twice. Could this be it's own pull request? Would that then allow backporting the #1212 fix to Foxy?

Reply:
> Could this be it's own pull request?

Yes, I actually should open an issue describing the problem.
I originally solved both here because they are related.

> Would that then allow backporting the #1212 fix to Foxy?

:+1:

Reply:
done, sorry for the delay